pep-0688.rst

```diff
@@ -0,0 +1,258 @@
+ask for the buffer. For some types, such as ``memoryview``,
+whether the buffer is writable depends on the instance:
+some instances are read-only and others are not. As such, we propose to
+support only whether a type implements the buffer protocol at
+all, not whether it supports more specific options such as
+writable buffers.
+
+Specification
+=============
+
+types.Buffer
+------------
+
+A new class ``types.Buffer`` will be added. It cannot be instantiated or
+subclassed, but supports the ``__instancecheck__`` and
+``__subclasscheck__`` hooks.  In CPython, these will check for the presence of the
+``bf_getbuffer`` slot in the type object:
+
+.. code-block:: pycon
+
+   >>> from types import Buffer
+   >>> isinstance(b"xy", Buffer)
+   True
+   >>> issubclass(bytes, Buffer)
+   True
+   >>> issubclass(memoryview, Buffer)
+   True
+   >>> isinstance("xy", Buffer)
+   False
+   >>> issubclass(str, Buffer)
+   False
+
+The new class can also be used in type annotations:
+
+.. code-block:: python
+
+   def need_buffer(b: Buffer) -> memoryview:
+       return memoryview(b)
+
+   need_buffer(b"xy")  # ok
+   need_buffer("xy")  # rejected by static type checkers
+
+Usage in stubs
+--------------
+
+For static typing purposes, types defined in C extensions usually
+require stub files, as described in :pep:`484`. In stub files,
+``types.Buffer`` may be used as a base class to indicate that a
+class implements the buffer protocol.
+
+For example, ``bytes`` may be declared as follows in a stub:
+
+.. code-block:: python
+
+   class bytes(types.Buffer, Sequence[int]):
+       def decode(self, ...): ...
+       ...
+
+The ``types.Buffer`` class does not require any special treatment
+in type checkers.
+
+Equivalent for older Python versions
+------------------------------------
+
+New typing features are usually backported to older Python versions
+in the ``typing_extensions`` package. Because the buffer protocol
+is accessible only in C, ``types.Buffer`` cannot be implemented
+in a pure Python package. As a temporary workaround, a
+``typing_extensions.Buffer`` ABC will be provided on Python versions
+that do not have ``types.Buffer`` available. For the benefit of
+static type checkers, ``typing_extensions.Buffer`` can be used as
+a base class in stubs to mark types as supporting the buffer protocol.
+For runtime uses, the ``ABC.register`` API can be used to register
+buffer classes with ``typing_extensions.Buffer``. When
+``types.Buffer`` is available, ``typing_extensions`` should simply
+re-export it.
+
+
+No special meaning for ``bytes``
+--------------------------------
+
+The special case stating that ``bytes`` may be used as a shorthand
+for other ``ByteString`` types will be removed from the ``typing``
+documentation.
+With ``types.Buffer`` available as an alternative, there is no good
+reason to allow ``bytes`` as a shorthand.
+We suggest that type checkers that implement this behavior should deprecate and
```

Comment:
Avoid double "that" and ensure it is clear that you're referring to the current behavior, not implementing new behavior specified in this PEP.

```suggestion
We suggest that type checkers currently implementing this behavior should deprecate and
```

or

```suggestion
We suggest that type checkers which currently implement this behavior should deprecate and
```